src/java/com/phenix/pct/PCTASBroker.java

```diff
@@ -0,0 +1,401 @@
+                }
+                if (uid.equalsIgnoreCase("auto")) {
+                    bw.println("uuid=" + new UID().toString());
+                } else if (!uid.equalsIgnoreCase("none")) {
+                    bw.println("uuid=" + uid);
+                }
+            }
+            bw.close();
+        } catch (IOException caught) {
+            throw new BuildException("Unable to create temp file", caught);
+        }
+    }
+
+    /**
+     * Creates an Exec task, calling mergeprop shell script.
+     */
+    private Task getCmdLineMergeTask(File propFile, File deltaFile) {
+        ExecTask task = new ExecTask(this);
+        task.setDir(getProject().getBaseDir());
+        task.setExecutable(getExecPath("mergeprop").getAbsolutePath());
+        task.setFailonerror(true);
+
+        Environment.Variable var1 = new Environment.Variable();
+        var1.setKey("DLC"); //$NON-NLS-1$
+        var1.setValue(getDlcHome().toString());
+        task.addEnv(var1);
+        Environment.Variable var2 = new Environment.Variable();
+        var2.setKey("JAVA_HOME"); //$NON-NLS-1$
+        var2.setValue(getJDK().getAbsolutePath());
+        task.addEnv(var2);
+        Environment.Variable var3 = new Environment.Variable();
+        var3.setKey("JREHOME"); //$NON-NLS-1$
+        var3.setValue(getJRE().getAbsolutePath());
+        task.addEnv(var3);
+
+        task.createArg().setValue("-type");
+        task.createArg().setValue("ubroker");
+        task.createArg().setValue("-action");
+        task.createArg().setValue(action);
+        task.createArg().setValue("-target");
+        task.createArg().setValue(propFile.getAbsolutePath());
+        task.createArg().setValue("-delta");
+        task.createArg().setValue(deltaFile.getAbsolutePath());
+
+        return task;
+    }
+
+    /**
+     * Cross-attributes check
+     * 
+     * @throws BuildException Attributes are wrong...
+     */
+    private void checkAttributes() {
+        if (action == null) {
+            throw new BuildException("Action attribute is missing");
+        }
+        if ((!UPDATE.equalsIgnoreCase(action)) && (!CREATE.equalsIgnoreCase(action))
+                && (!DELETE.equalsIgnoreCase(action))) {
+            throw new BuildException("Unknown action : " + action);
+        }
+        if ((brokerLogLevel != -1) && ((brokerLogLevel < 1) || (brokerLogLevel > 5))) {
+            throw new BuildException("Log level should be between 1 and 5");
+        }
+        if ((serverLogLevel != -1) && ((serverLogLevel < 1) || (brokerLogLevel > 5)))
+            throw new BuildException("Log level should be between 1 and 5");
+
+        if (name == null) {
+            throw new BuildException("Name attribute is missing");
+        }
+        if (name.trim().length() == 0) {
+            throw new BuildException("Name attribute is empty");
+        }
+        // FIXME Operating mode is case-sensitive, so we should always replace value with the correct one
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1134)